### ISSUE TITLE
Gradient implementation of Dice loss function

### ISSUE THREAD

**Issue by amh28** [NifTK/NiftyNet#27]
Hello everyone, I have been reviewing the Dice loss for segmentation implemented in this platform, but I could not find the gradient implementation found in the original paper[1] which is the derivation of the Dice loss: 

![gradient](https://user-images.githubusercontent.com/4590737/35054055-bc7a9cd2-fb79-11e7-844a-bf1f377958b2.png)
[1]  V-Net: Fully Convolutional Neural Networks for Volumetric Medical Image Segmentation

Also, on application/segmentation_application.py it directly uses the self.optimiser.compute_gradients(loss) function, so is this derivation done automatically? Any insights on this would be very appreciated.

**Comment by wyli** [comment]
Hi @amh28  yes, that's done automatically by `compute_gradients()` as you described.
